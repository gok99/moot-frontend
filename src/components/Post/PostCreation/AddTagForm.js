import React, { useState, useRef } from 'react';
import { Row, Col, Form, Button, Overlay, Tooltip } from 'react-bootstrap';

import '../../Access/access.css'

const AddTagForm = ({ tagList, onAddTag, postCreationCheck, ownedError }) => {
  const [currentTag, setCurrentTag] = useState("");
  const target = useRef(null);
  const [error, setError] = useState(ownedError);
  const tags = tagList.map((tag) => {
    return <option>{tag}</option>
  });

  const btnTags = typeof InstallTrigger !== 'undefined'
    ? tagList.map((tag) => {
        return <option onClick={onAddTag(tag)}>{tag}</option>
      })
    : tagList.map((tag) => {
        return <option value={tag}>{tag}</option>
      });

  const onChange = (event) => {
    setCurrentTag(event.target.value);
  };

  const onChangeTwo = (event) => {
    setCurrentTag(event.target.value);
    onAddTag(event.target.value);
  };

  return postCreationCheck 
    ? (
      <Row className="mt-2 mb-2" >
        <Col xs={8}>
          <Form.Control as="select" value={currentTag} onChange={onChange}>
            <option>&lt;None&gt;</option>
            {tags}
          </Form.Control>
        </Col>
        <Col xs={4}>
          <Button ref={target} className="btn-postcreation btn-addtag form" onClick={onAddTag(currentTag)}>
            Add Tag
          </Button>
          <Overlay target={target} show={!!ownedError} placement="right">
            {(props) => (
              <Tooltip id="tooltip-access" {...props}>
                {ownedError}
              </Tooltip>
            )}
          </Overlay>
        </Col>
      </Row>
      )
    : (
      <Row className="mt-2 mb-2" >
        <Col xs={8}>
          <Button className="btn-selection" onClick={onAddTag("<Home>")}>
            All
          </Button>
        </Col>
        <Col xs={4}>
          { typeof InstallTrigger !== 'undefined'
            ? <Form.Control as="select" value={currentTag} onChange={onChange}>
                <option>&lt;Home&gt;</option>
                {btnTags}
              </Form.Control>
            : <Form.Control as="select" value={currentTag} onChange={onChangeTwo}>
                <option>&lt;Home&gt;</option>
                {btnTags}
              </Form.Control>
          }
        </Col>
      </Row>
      );
};

export default AddTagForm;