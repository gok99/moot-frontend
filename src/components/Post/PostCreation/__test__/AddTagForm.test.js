import React from 'react';
import ReactDOM from 'react-dom';
import AddTagForm from '../AddTagForm';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddTagForm tagList={[]} onAddTag={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} libraryCheck={true} ownedError={""} currentPostTag={""} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddTagForm tagList={[]} onAddTag={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} libraryCheck={false} ownedError={""} currentPostTag={""} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddTagForm tagList={[]} onAddTag={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={false} libraryCheck={false} ownedError={""} currentPostTag={""} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<AddTagForm tagList={[]} onAddTag={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} libraryCheck={true} ownedError={""} currentPostTag={""} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<AddTagForm tagList={[]} onAddTag={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} libraryCheck={false} ownedError={""} currentPostTag={""} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<AddTagForm tagList={[]} onAddTag={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={false} libraryCheck={false} ownedError={""} currentPostTag={""} />).toJSON();
  expect(tree).toMatchSnapshot();
});