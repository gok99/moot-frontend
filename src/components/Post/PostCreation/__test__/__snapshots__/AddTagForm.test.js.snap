// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-2 mb-2 row"
>
  <div
    className="col-8"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value=""
    >
      <option>
        &lt;None&gt;
      </option>
    </select>
  </div>
  <div
    className="col-4"
  >
    <button
      className="btn-postcreation btn-addtag form btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Click here to Add!
    </button>
  </div>
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="mt-2 mb-2 row"
>
  <div
    className="col-12"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        &lt;None&gt;
      </option>
    </select>
  </div>
</div>
`;

exports[`matches snapshot 3`] = `
<div
  className="mt-2 mb-2 row"
>
  <div
    className="col-4"
  >
    <button
      className="btn-selection btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      All
    </button>
  </div>
  <div
    className="d-flex justify-content-end col-4"
  >
    <p
      className="text-post sub"
    >
      Currently viewing "All":
    </p>
  </div>
  <div
    className="col-4"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value=""
    >
      <option
        value="All"
      >
        All
      </option>
    </select>
  </div>
</div>
`;
